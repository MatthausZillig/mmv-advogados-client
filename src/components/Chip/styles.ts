import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.xxsmall};
    margin: 0px 8px 8px 0px;
    padding: 4px 8px;
    background: ${theme.colors.secondary};
    border-radius: 4px;
    width: auto;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`
