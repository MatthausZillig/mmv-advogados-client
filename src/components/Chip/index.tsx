import * as S from './styles'

export type ChipProps = {
  children?: React.ReactNode
}

const Chip = ({ children }: ChipProps) => (
  <S.Wrapper data-testid="chip">{children}</S.Wrapper>
)

export default Chip
