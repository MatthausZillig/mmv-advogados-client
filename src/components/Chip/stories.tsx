import { Story, Meta } from '@storybook/react/types-6-0'
import Chip, { ChipProps } from '.'

export default {
  title: 'Chip',
  component: Chip
} as Meta

export const Default: Story<ChipProps> = () => (
  <div style={{ display: 'flex', width: '20rem' }}>
    <Chip>Chip</Chip>
  </div>
)
