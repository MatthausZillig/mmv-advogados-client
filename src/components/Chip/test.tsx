import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Chip from '.'

describe('<Chip />', () => {
  it('should render the chip', () => {
    const { container } = renderWithTheme(<Chip>Chip</Chip>)

    expect(screen.getByTestId('chip')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
