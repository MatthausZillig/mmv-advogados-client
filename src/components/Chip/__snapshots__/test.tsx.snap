// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> should render the chip 1`] = `
.c0 {
  color: #FCFCFC;
  font-size: 1rem;
  margin: 0px 8px 8px 0px;
  padding: 4px 8px;
  background: #0E5BDF;
  border-radius: 4px;
  width: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
  data-testid="chip"
>
  Chip
</div>
`
